payjoin/src/v2.rs

```diff
@@ -0,0 +1,172 @@
+use std::{error, fmt};
+
+pub const MAX_BUFFER_SIZE: usize = 65536;
+pub const RECEIVE: &str = "receive";
+pub const PADDED_MESSAGE_BYTES: usize = 7168; // 7KB
+
+pub fn subdir(path: &str) -> String {
+    let subdirectory: String;
+
+    if let Some(pos) = path.rfind('/') {
+        subdirectory = path[pos + 1..].to_string();
+    } else {
+        subdirectory = path.to_string();
+    }
+
+    let pubkey_id: String;
+
+    if let Some(pos) = subdirectory.find('?') {
+        pubkey_id = subdirectory[..pos].to_string();
+    } else {
+        pubkey_id = subdirectory;
+    }
+    pubkey_id
+}
+
+use bitcoin::secp256k1::ecdh::SharedSecret;
+use bitcoin::secp256k1::{PublicKey, Secp256k1, SecretKey};
+use chacha20poly1305::aead::{Aead, KeyInit, OsRng, Payload};
+use chacha20poly1305::{AeadCore, ChaCha20Poly1305, Nonce};
+
+/// crypto context
+///
+/// <- Receiver S
+/// -> Sender E, ES(payload), payload protected by knowledge of receiver key
+/// <- Receiver E, EE(payload), payload protected by knowledge of sender & receiver key
+pub fn encrypt_message_a(
```

Comment:
half of these are used only in the send module, half in recieve, they should be tagged with conditional compilation based on send / receive features